File: sdbg/src/main/java/org/cf/sdbg/command/ClearScreen.java
package org.cf.sdbg.command;

import java.io.IOException;
import java.util.concurrent.Callable;
import picocli.CommandLine;

/**
 * Command that clears the screen.
 */
@CommandLine.Command(name = "cls", aliases = "clear", mixinStandardHelpOptions = true, version = "1.0",
    description = "Clears the screen")
public class ClearScreen implements Callable<Void> {

  @CommandLine.ParentCommand
  CliCommands parent;

  public Void call() throws IOException {
    parent.reader.clearScreen();
    return null;
  }
}

File: sdbg/src/main/java/org/cf/sdbg/command/ClearScreen.kt
package org.cf.sdbg.command

import picocli.CommandLine
import picocli.CommandLine.ParentCommand
import java.io.IOException
import java.util.concurrent.Callable

/**
 * Command that clears the screen.
 */
@CommandLine.Command(name = "cls", aliases = ["clear"], mixinStandardHelpOptions = true, version = ["1.0"], description = ["Clears the screen"])
class ClearScreen : Callable<Void?> {
    @ParentCommand
    var parent: CliCommands? = null

    @Throws(IOException::class)
    override fun call(): Void? {
        parent!!.reader!!.clearScreen()
        return null
    }
}
